packages/react-reconciler/src/ReactFiberCompleteWork.new.js

```diff
@@ -1084,32 +1089,40 @@ function completeWork(
         }
       }
 
-      if (supportsPersistence) {
-        // TODO: Only schedule updates if not prevDidTimeout.
-        if (nextDidTimeout) {
-          // If this boundary just timed out, schedule an effect to attach a
-          // retry listener to the promise. This flag is also used to hide the
-          // primary children.
-          workInProgress.flags |= Update;
-        }
+      const wakeables: Set<Wakeable> | null = (workInProgress.updateQueue: any);
+      if (wakeables !== null) {
+        // Schedule an effect to attach a retry listener to the promise.
+        // TODO: Move to passive phase
+        workInProgress.flags |= Update;
       }
+
       if (supportsMutation) {
-        // TODO: Only schedule updates if these values are non equal, i.e. it changed.
-        if (nextDidTimeout || prevDidTimeout) {
-          // If this boundary just timed out, schedule an effect to attach a
-          // retry listener to the promise. This flag is also used to hide the
-          // primary children. In mutation mode, we also need the flag to
-          // *unhide* children that were previously hidden, so check if this
-          // is currently timed out, too.
-          workInProgress.flags |= Update;
+        if (nextDidTimeout !== prevDidTimeout) {
+          // In mutation mode, visibility is toggled by mutating the nearest
+          // host nodes whenever they switch from hidden -> visible or vice
+          // versa. We don't need to switch when the boundary updates but its
+          // visibility hasn't changed.
+          workInProgress.flags |= Visibility;
         }
       }
+      if (supportsPersistence) {
+        if (nextDidTimeout) {
```

Comment:
As I read this, I wondered why we didn't also have the `nextDidTimeout !== prevDidTimeout` check here, then I noticed before we also had this comment that didn't get copied over:
```js
// TODO: Only schedule updates if not prevDidTimeout.
```
I'm not really familiar with the intricacies of persistent mode. Was this TODO just not necessary? Or was this a copy-paste error?

Reply:
It's still relevant but I replaced it with the TODO comment below: 

```js
          // In persistent mode, visibility is toggled by cloning the nearest
          // host nodes in the complete phase whenever the boundary is hidden.
          // TODO: The plan is to add a transparent host wrapper (no layout)
          // around the primary children and hide that node. Then we don't need
          // to do the funky cloning business.
```